impl/src/main/java/org/jboss/weld/util/ServiceLoader.java

```diff
@@ -232,27 +232,18 @@ private void loadService(String serviceClassName, URL file, int lineNumber) {
         return serviceClass;
     }
 
-    private S prepareInstance(Class<? extends S> serviceClass) {
-        try {
-            // TODO Support the SM
-            Constructor<? extends S> constructor = serviceClass.getDeclaredConstructor();
-            constructor.setAccessible(true);
-            return constructor.newInstance();
-        } catch (LinkageError e) {
```

Comment:
Hm, I think we should follow the ServiceLoader from JDK, i.e. catch `Throwable` and throw `ServiceConfigurationError`?